### DISCUSSION TITLE
Rendering response from up.request

### DISCUSSION THREAD

**Discussion by tmbenhura** [unpoly/unpoly#282] [Q&A]
How do I render the result of up.request?

I am intercepting a form submission to add custom FormData, even though I'm setting a body target, when the response is complete the page is not being updated with the result

**Comment by adam12** [comment]
Can you share what you have so far?

**Reply by tmbenhura** [reply]
`
      let formData = new FormData();

      formData.append('file', blob, 'filter-applied.jpg');
      
      up.request($form.attr('action'),
        {
          params: formData,
          method: 'POST',
          target: 'body'
        })
`

**Reply by adam12** [reply]
Perhaps Henning will have a better suggestion, but what about replacing `up.request` with [`up.render`](https://unpoly.com/up.render)?

**Reply by tmbenhura** [reply]
The image is successfully uploaded and the server returns valid (200) response from the page the request redirects to.

**Reply by adam12** [reply]
> The image is successfully uploaded and the server returns valid (200) response from the page the request redirects to.

Was the `{ target }` updated correctly? I'm guessing no?

**Reply by tmbenhura** [reply]
up.render works, thanks @adam12 . I initially had problems with it setting up the URL vis-a-vis the documented example. I thought since, up.request had a "target" parameter is was also capable of rendering.

`
      up.render('body', {
        url: $form.attr('action'),
        params: formData,
        method: 'POST'
      })
`

**Reply by adam12** [reply]
So `up.render({target: 'body', ...reset of bits...})` didn't work?

Interesting. Maybe it's a bug or a documentation issue. I'll have to look.

**Reply by tmbenhura** [reply]
No, it didn't. I'm using version 2.1, at the moment (I experienced some transpilation issues with 2.3, that I'm yet to revisit)
